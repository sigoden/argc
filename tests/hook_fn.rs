use crate::*;

#[test]
fn hook_without_subcmd() {
    let script = r###"
_argc_before() { echo before; }
_argc_after() { echo after; }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn hook_with_main() {
    let script = r###"
_argc_before() { echo before; }
_argc_after() { echo after; }
main() { :; }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn hook_with_subcmd() {
    let script = r###"
_argc_before() { echo before; }
_argc_after() { echo after; }
# @cmd
cmd() { :; }
"###;
    snapshot!(script, &["prog", "cmd"]);
}

#[test]
fn hook_only_before() {
    let script = r###"
_argc_before() { echo before; }
main() { :; }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn hook_only_after() {
    let script = r###"
_argc_after() { echo after; }
main() { :; }
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn hook_param_fn() {
    let script = r###"
_argc_before() { echo before; }
_argc_after() { echo after; }
_choice_fn() { :; }
"###;
    snapshot!(script, &["prog", "___internal___", "_choice_fn"]);
}
