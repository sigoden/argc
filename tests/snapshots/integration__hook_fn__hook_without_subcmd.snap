---
source: tests/hook_fn.rs
expression: data
---
RUN
prog

# OUTPUT
argc__args=( prog )
argc__positionals=(  )
_argc_before

# BUILD_OUTPUT
before


